### ISSUE TITLE
Types for Apparent Power and Reactive Power

### ISSUE THREAD

**Issue by madebyjeffrey** [anjdreas/UnitsNet#185]
In engineering fields that use electricity and motors or inductors it is common to have to consider apparent power (Volt Amps) and Reactive Power (vars  - volts reactive).

You will see volt amps (VA) on UPS equipment as well.

Quite related to this concept is "power factor" which is unitless - are there provisions for unit less values that we still want to understand as representing something specific?

**Comment by anjdreas** [comment]
I'm not too familiar with these units. Are you suggesting to add the following unit types (class) and corresponding units of measurement (values/conversions)?

* ApparentPower
  * VoltAmps
* ReactivePower
  * Vars
* PowerFactor
  * Factor

I did a quick google and it seems to me that there is only a single unit of measurement for each of these, meaning it does not make sense to convert VoltAmps to a different unit of measurement, in the way we convert a length in meters to centimeters. Is that correct?

We absolutely can add unit types with only a single unit of measurement. You get static typing of values in the code to avoid passing watts or volts into methods expecting apparent power, but you miss out on any conversions. We haven't had this issue up until now, so this is a new thing to consider for me. What do you think? Is it worth having the unit types, even if they don't give you any conversions?

PowerFactor could possibly be replaced by [Ratio](https://github.com/anjdreas/UnitsNet/blob/master/UnitsNet/GeneratedCode/UnitClasses/Ratio.g.cs), which is also dimensionless. I can imagine creating operator overloads giving you this `Ratio powerFactor = Power.FromWatts(100) / ApparentPower.FromVoltAmps(10);`
Since there is a huge variety of ratios/factors in this world, I don't think it is a good idea to start adding unit types for each of them, just for the sake of type safety. So I propose not adding PowerFactor as a unit type, but instead use Ratio. Thoughts?

**Comment by anjdreas** [comment]
Do you have any comments back on this?

**Comment by madebyjeffrey** [comment]
Sorry for the delay on this. It could be handy to be have some for applications involving power. I could imagine them being useful in applications involving the conversion with inductance and capacitance with power.

Although my own use now requires a strictly run time application, rather than a compiler type check.

**Comment by anjdreas** [comment]
I see. I think this is one of the cases where I am not sure whether it makes the most sense to add a separate nuget, say `UnitsNet.ElectricalEngineering`, or just add it to the core. If there are a ton of other units that would go into that nuget, then it would make sense, but just for 2-3 units? Not sure.

What do you think?

**Comment by JKSnd** [comment]
This [PR](https://github.com/anjdreas/UnitsNet/pull/197), that we decided not to merge, might be relevant to this discussion.

TL;DR: I want to create a representation of AC Voltage that would expose properties such as Frequency, Phase, Waveform (sinusoid, triangle, etc), and Amplitude metrics (Peak, RMS, etc). Maybe we should talk about creating such an EE toolkit, @madebyjeffrey.

**Comment by JKSnd** [comment]
@madebyjeffrey, I've started another project that will basically satisfy the needs of the [PR](https://github.com/anjdreas/UnitsNet/pull/197) I mentioned. I've called it [Signals](https://github.com/PkSound/Signals), and although its meant to represent signals for use in an audio domain, there's no reason that we couldn't extend it to include power concepts. The differences are just in scale.

**Comment by anjdreas** [comment]
Closing this in favor of #226. Continue the conversation there.
